packages/react/src/primitives/FieldGroupIcon/FieldGroupIcon.tsx

```diff
@@ -1,19 +1,27 @@
+import * as React from 'react';
 import classNames from 'classnames';
 
 import { ComponentClassNames } from '../shared/constants';
-import { FieldGroupIconProps, Primitive } from '../types';
+import { FieldGroupIconProps, PrimitiveWithForwardRef } from '../types';
 import { View } from '../View';
 
-export const FieldGroupIcon: Primitive<FieldGroupIconProps, 'button'> = ({
-  className,
-  children,
-  isVisible = true,
-  excludeFromTabOrder = false,
-  ...rest
-}) => {
+const FieldGroupIconInner: PrimitiveWithForwardRef<
+  FieldGroupIconProps,
+  'button' | 'div'
```

Comment:
What's the reason to change `'button'` to `'button' | 'div'`?

Reply:
This one was tricky to type because the `FieldGroupIcon` can either be a `div` in the case of an non-interactive icon, or a `button` in the case of a clickable icon. Example here:
https://ui.docs.amplify.aws/components/textfield#inner-components-icons

![Screen Shot 2021-11-15 at 11 12 15 PM](https://user-images.githubusercontent.com/6165315/141931459-57cb2449-0bac-491e-ad3f-174529c40c84.png)